src/client.rs

```diff
@@ -277,6 +282,23 @@ pub unsafe extern "C" fn native_stats_cb<C: Context>(
     1 // librdkafka will not free the json buffer
 }
 
+pub unsafe extern "C" fn native_error_cb<C: Context>(
+        _client: *mut RDKafka, err: i32, reason: *const i8,
+        opaque: *mut c_void) {
+    let err = rdsys::primive_to_rd_kafka_resp_err_t(err).expect("global error not an rd_kafka_resp_err_t");
+    let error = match err {
+        rdsys::rd_kafka_resp_err_t::RD_KAFKA_RESP_ERR__ALL_BROKERS_DOWN => {
+            KafkaError::AllBrokersDown(err)
```

Comment:
It's actually pretty hard to match on, you need to get into the `rd_kafka_resp_err_t` values in rdsys. We did a similar thing with partition eof.

Maybe we should keep a mapping to a nice Rust enum of all values including documentation?